src/gz/bufread.rs

```diff
@@ -449,9 +453,13 @@ impl<R: BufRead> MultiGzDecoder<R> {
 }
 
 impl<R> MultiGzDecoder<R> {
-    /// Returns the current header associated with this stream, if it's valid
-    pub fn header(&self) -> Option<&GzHeader> {
```

Comment:
I think there is no need to make this a breaking change by removing the single-header access method. It can live side-by-side with the `headers()` method.